lib/enums/gradient_direction.dart

```diff
@@ -1,11 +1,83 @@
-enum GradientDirection {
-  topLeft,
-  topCenter,
-  topRight,
-  centerLeft,
-  center,
-  centerRight,
-  bottomLeft,
-  bottomCenter,
-  bottomRight
+import 'package:flutter/material.dart';
+
+sealed class GradientDirection {
+  const GradientDirection();
+
+  static const topLeft = GradientDirectionTopLeft._();
+  static const topCenter = GradientDirectionTopCenter._();
+  static const topRight = GradientDirectionTopRight._();
+  static const centerLeft = GradientDirectionCenterLeft._();
+  static const center = GradientDirectionCenter._();
+  static const centerRight = GradientDirectionCenterRight._();
+  static const bottomLeft = GradientDirectionBottomLeft._();
+  static const bottomCenter = GradientDirectionBottomCenter._();
+  static const bottomRight = GradientDirectionBottomRight._();
+
+  factory GradientDirection.custom({
+    Alignment alignment = const Alignment(-0.5, 0),
+    Alignment endAlignment = const Alignment(0.5, 0),
+  }) {
+    return GradientDirectionCustom._(
+      alignment: alignment,
+      endAlignment: endAlignment,
+    );
+  }
```

Comment:
Currently, tapping on "Custom" resets the gradient direction to the default values for `GradientDirection.custom`. 

https://github.com/victoreronmosele/flutter_gradient_generator/assets/19398044/8ef31f51-0289-443f-8fdf-02f9a5babb49

It will be nice to still keep the current gradient direction even after tapping on "Custom" so the user can continue with their current direction. 

A use case is if a user just wants to adjust one of the preset directions.

What do you think?

Reply:
this is a great idea! Totally doable. Gonna implement it.